### PR TITLE
coll/sm: fix minor memory leak

### PR DESCRIPTION
**Pull request by ggouaillardet**
this is a backport of commit open-mpi/ompi@5d44a30111e730c1a6a9aef19548b87e704316de in master

### CODE REVIEW

**Comment by jsquyres** [comment]
@ggouaillardet Who do you want to review this PR?

**Comment by ggouaillardet** [comment]
@jsquyres can you review it ?

File: ompi/mca/coll/sm/coll_sm_module.c [hunk 1]
```diff
@@ -99,6 +101,7 @@ static void mca_coll_sm_module_destruct(mca_coll_sm_module_t *module)
             /* Ignore any errors -- what are we going to do about
                them? */
             mca_common_sm_fini(c->sm_bootstrap_meta);
+            OBJ_RELEASE(c->sm_bootstrap_meta);
```
Comment (jsquyres) [inline ompi/mca/coll/sm/coll_sm_module.c:104]: In a first-order review, I don't see module->cm_bootstrap_meta being set to NULL in the constructor.

Without that, how can you know it's safe to OBJ_RELEASE c->sm_bootstream_beta?

File: ompi/mca/coll/sm/coll_sm_module.c [hunk 2]
```diff
@@ -611,6 +614,7 @@ static int bootstrap_comm(ompi_communicator_t *comm,
         mca_common_sm_init_group(comm->c_local_group, size, fullpath,
                                  sizeof(mca_common_sm_seg_header_t),
                                  getpagesize());
+    free(fullpath);
```
Comment (jsquyres) [inline ompi/mca/coll/sm/coll_sm_module.c:617]: This looks correct.

File: ompi/mca/coll/sm/coll_sm_module.c (hunk 1 above)
Comment (ggouaillardet) [inline ompi/mca/coll/sm/coll_sm_module.c:104]: c is **not** an object but is malloc'ed in ompi_coll_sm_lazy_enable(...)

sm_bootstrap_meta will be initialized in bootstrap_comm (invoked from ompi_coll_sm_lazy_enable), unless bootstrap_comm fails.
but even this case is correctly handled by ompi_coll_sm_lazy_enable.

also, if c->sm_bootstrap_meta were not a vaild pointer, mca_common_sm_fini would have crashed first with a glory SIGSEGV

Comment (jsquyres) [inline ompi/mca/coll/sm/coll_sm_module.c:104]: Cool.

**Pull request merged**